src/Datadog.Trace/Logging/DatadogLogging.cs

```diff
@@ -133,6 +132,10 @@ static DatadogLogging()
                 Initialized = true;
                 // nothing else to do here
             }
+            finally
+            {
+                CoreLogStrategy.SetStrategy(GetSharedLogger);
```

Comment:
This can probably be the first thing that happens actually, as it will be lazily called for the classes that need it.